pycbc/inference/recal.py

```diff
@@ -0,0 +1,102 @@
+        init_detuning = self.freq**2 / (self.freq**2 - 1.0j * self.freq * \
+                                        self.fs0 * self.qinv0 + self.fs0**2)
+
+        # initial open loop gain
+        self.g0 = self.c0 * self.d0 * (self.a_tst0 + self.a_pu0)
+
+        # initial response function
+        self.r0 = (1.0 + self.g0) / self.c0
+
+        # residual of c0 after factoring out the coupled cavity pole fc0
+        self.c_res = self.c0 * (1 + 1.0j * self.freq / self.fc0) / init_detuning
+
+    def update_c(self, fs=None, qinv=None, fc=None, kappa_c=1.0):
+        detuning_term = self.freq**2 / (self.freq**2 - 1.0j * self.freq * fs * \
+                                        qinv + fs**2)
+        return self.c_res * kappa_c / (1 + 1.0j * self.freq/fc) * detuning_term
+
+    def update_g(self, fs=None, qinv=None, fc=None, kappa_tst_re=1.0,
+                 kappa_tst_im=0.0, kappa_pu_re=1.0, kappa_pu_im=0.0,
+                 kappa_c=1.0):
+        c = self.update_c(fs=fs, qinv=qinv, fc=fc, kappa_c=kappa_c)
+        a_tst = self.a_tst0 * (kappa_tst_re + 1.0j * kappa_tst_im)
+        a_pu = self.a_pu0 * (kappa_pu_re + 1.0j * kappa_pu_im)
+        return c * self.d0 * (a_tst + a_pu)
+
+    def update_r(self, fs=None, qinv=None, fc=None, kappa_c=1.0,
+                 kappa_tst_re=1.0, kappa_tst_im=0.0, kappa_pu_re=1.0,
+                 kappa_pu_im=0.0):
+        c = self.update_c(fs=fs, qinv=qinv, fc=fc, kappa_c=kappa_c)
+        g = self.update_g(fs=fs, qinv=qinv, fc=fc, kappa_c=kappa_c,
+                          kappa_tst_re=kappa_tst_re, kappa_tst_im=kappa_tst_im,
+                          kappa_pu_re=kappa_pu_re, kappa_pu_im=kappa_pu_im)
+        return (1.0 + g) / c
+
+    def adjust_strain(self, strain, params):
+        """Adjust the FrequencySeries strain
+        """
+
+        fs = params["calib_fs"] if "calib_fs" in params else self.fs0
+        qinv = params["calib_qinv"] if "calib_qinv" in params else self.qinv0
+        fc = self.fc0+params["calib_deltafc"] if "calib_deltafc" in params \
+             else self.fc0
+        kappa_c = params["calib_kappa_c"] if "calib_kappa_c" in params else 1.0
+        kappa_tst_re = params["calib_kappa_tst_re"] if "calib_kappa_tst_re" in \
+                       params else 1.0
+        kappa_tst_im = params["calib_kappa_tst_im"] if "calib_kappa_tst_im" in \
+                       params else 0.0
+        kappa_pu_re = params["calib_kappa_pu_re"] if "calib_kappa_pu_re" in \
+                      params else 1.0
+        kappa_pu_im = params["calib_kappa_pu_im"] if "calib_kappa_pu_im" in \
+                      params else 0.0
+
+        r_adjusted = self.update_r(fs=fs, qinv=qinv, fc=fc, kappa_c=kappa_c,
+                                   kappa_tst_re=kappa_tst_re,
+                                   kappa_tst_im=kappa_tst_im,
+                                   kappa_pu_re=kappa_pu_re,
+                                   kappa_pu_im=kappa_pu_im)
+
```

Comment:
Okay, check out: https://github.com/cmbiwer/pycbc-cal/blob/master/pycbc_cal/cal.py

I had my own class to do this and its gives you an idea of the kind of documentation to add - and I wrote most of it so it should be mostly copy and paste.